src/merkle/smt/mod.rs

```diff
@@ -43,13 +43,13 @@ pub const SMT_MAX_DEPTH: u8 = 64;
 /// must accomodate all keys that map to the same leaf.
 ///
 /// [SparseMerkleTree] currently doesn't support optimizations that compress Merkle proofs.
-pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
+pub trait SparseMerkleTree<const DEPTH: u8> {
     /// The type for a key
-    type Key: Clone;
+    type Key: Clone + Ord;
```

Comment:
That's explained in the commit message https://github.com/reilabs/miden-crypto/commit/1cea08c2ff16910c4fd7321eb1568d0fdaa653e0:

> As part of this change, SparseMerkleTree now requires its Key to be Ord and its Leaf to be Clone (both bounds which were already met by existing implementations). The Ord bound could instead be changed to Eq + Hash, if MutationSet were changed to use a HashMap instead of a BTreeMap.